src/bscPlugin/codeActions/CodeActionsProcessor.ts

```diff
@@ -0,0 +1,108 @@
+import { CodeActionKind } from 'vscode-languageserver';
+import { codeActionUtil } from '../../CodeActionUtil';
+import type { DiagnosticMessageType } from '../../DiagnosticMessages';
+import { DiagnosticCodeMap } from '../../DiagnosticMessages';
+import type { BrsFile } from '../../files/BrsFile';
+import type { XmlFile } from '../../files/XmlFile';
+import type { OnGetCodeActionsEvent } from '../../interfaces';
+import { util } from '../../util';
+
+export class CodeActionsProcessor {
+    public constructor(
+        public event: OnGetCodeActionsEvent
+    ) {
+
+    }
+
+    public process() {
+        for (const diagnostic of this.event.diagnostics) {
+            if (diagnostic.code === DiagnosticCodeMap.callToUnknownFunction) {
+                this.suggestImports(diagnostic as any);
```

Comment:
Can do better than `as any`?

Reply:
We can be more specific, but the `suggestImports` function already typed the diagnostic parameter. It just seemed needlessly verbose to cast as the same type when the entire loop is clearly related to diagnostics. 

```typescript
private suggestImports(diagnostic: DiagnosticMessageType<'callToUnknownFunction'>) {
```